plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -76,39 +86,77 @@ export const useRoles = (
     permission: policyEntityUpdatePermission,
     resourceRef: policyEntityUpdatePermission.resourceType,
   });
+
+  React.useEffect(() => {
+    const fetchAllPermissionPolicies = async () => {
+      if (!Array.isArray(roles)) return;
+      const conditionPromises = roles.map(async role => {
+        try {
+          const conditionalPolicies = await rbacApi.getRoleConditions(
+            role.name,
```

Comment:
@ciiay you need to manually throw the error here because of you see the api file the apis doenst throw error so we are returning the error response. You need to use that error response to get the error msg and throw it manually.

```
if((conditionaPolicies as unknown as Response)?.statusText) throw new Error((conditionaPolicies as unknown as Response).statusText);
```